lib/motsu/src/context/mod.rs

```diff
@@ -0,0 +1,188 @@
+}
+
+impl Context {
+    /// Get test context associated with the current test thread.
+    #[must_use]
+    pub fn current() -> Self {
+        Self { thread_name: ThreadName::current() }
+    }
+
+    /// Get the value at `key` in storage.
+    pub(crate) fn get_bytes(self, key: &Bytes32) -> Bytes32 {
+        let context = EVM.entry(self.thread_name).or_default();
+        context.storage.contract_data.get(key).copied().unwrap_or_default()
+    }
+
+    /// Get the raw value at `key` in storage and write it to `value`.
+    pub(crate) unsafe fn get_bytes_raw(self, key: *const u8, value: *mut u8) {
+        let key = read_bytes32(key);
+
+        write_bytes32(value, self.get_bytes(&key));
+    }
+
+    /// Set the value at `key` in storage to `value`.
+    pub(crate) fn set_bytes(self, key: Bytes32, value: Bytes32) {
+        let mut context = EVM.entry(self.thread_name).or_default();
+        context.storage.contract_data.insert(key, value);
+    }
+
+    /// Set the raw value at `key` in storage to `value`.
+    pub(crate) unsafe fn set_bytes_raw(self, key: *const u8, value: *const u8) {
+        let (key, value) = (read_bytes32(key), read_bytes32(value));
+        self.set_bytes(key, value);
+    }
+
+    /// Clears storage, removing all key-value pairs associated with the current
+    /// test thread.
+    pub fn reset_storage(self) {
+        EVM.remove(&self.thread_name);
+    }
+
+    /// Gets the code hash of the account at the given address.
+    pub fn account_codehash(self) -> [u8; 66] {
+        let context = EVM.entry(self.thread_name).or_default();
+        context.environment.account_codehash()
+    }
+
+    /// Gets a bounded estimate of the Unix timestamp at which the Sequencer
+    /// sequenced the transaction.
+    pub fn block_timestamp(self) -> u64 {
+        let context = EVM.entry(self.thread_name).or_default();
+        context.environment.block_timestamp()
+    }
+
+    /// Gets the chain ID of the current chain.
+    pub fn chain_id(self) -> u64 {
+        let context = EVM.entry(self.thread_name).or_default();
+        context.environment.chain_id()
+    }
+
+    /// Gets the address of the current program.
+    pub fn contract_address(self) -> [u8; 42] {
+        let context = EVM.entry(self.thread_name).or_default();
+        context.environment.contract_address()
+    }
+
+    /// Emits an EVM log with the given number of topics and data, the first
+    /// bytes of which should be the 32-byte-aligned topic data.
+    ///
+    /// Data contains `topics` amount of topics and then encoded event in `data`
+    /// buffer.
+    pub(crate) unsafe fn emit_log(
+        self,
+        data: *const u8,
+        len: usize,
+        topics: usize,
+    ) {
+        // https://github.com/OffchainLabs/stylus-sdk-rs/blob/v0.6.0/stylus-sdk/src/evm.rs#L38-L52
+        let buffer = read_bytes(data, len);
+        let encoded_event: Vec<u8> =
+            buffer.clone().into_iter().skip(topics * WORD_BYTES).collect();
```

Comment:
If you would have `Event` struct (or `EncodedEvent`), here you could also add to it conversion from `buffer`